### PR TITLE
MOD-6344 Fix potential crash for cf.scandump and cf.loadchunk

### PR DESCRIPTION
**Pull request by tezc**
Add boundary checks for cf.scandump and cf.loadchunk.

### CODE REVIEW

**Review by LiorKogan — DISMISSED** [review]

File: src/cf.c [hunk 1]
```diff
@@ -84,7 +84,7 @@ const char *CF_GetEncodedChunk(const CuckooFilter *cf, long long *pos, size_t *b
 }
 
 int CF_LoadEncodedChunk(const CuckooFilter *cf, long long pos, const char *data, size_t datalen) {
-    if (datalen == 0) {
+    if (datalen == 0 || pos < datalen + 1) {
```
Comment (LiorKogan) [inline src/cf.c:87]: Please add protection for datalen = -1

File: src/cf.c [hunk 2]
```diff
@@ -102,6 +102,11 @@ int CF_LoadEncodedChunk(const CuckooFilter *cf, long long pos, const char *data,
         offset -= currentSize;
     }
 
+    // Boundary check before memcpy()
+    if (!filter || filter->bucketSize * filter->numBuckets < offset + datalen) {
```
Comment (LiorKogan) [inline src/cf.c:106]: Consider adding checks for protections against overflows

File: src/cf.c (hunk 1 above)
Comment (tezc) [reply src/cf.c:87]: It's type `size_t`, cannot be negative.

**Review by YaacovHazan — APPROVED** [review]

File: src/cf.c (hunk 1 above)
Comment (LiorKogan) [reply src/cf.c:87]: can be max positive (2^32-1) so (2^32-1+1 is 0)

**Review by LiorKogan — APPROVED** [review]

**Pull request merged**